### PR TITLE
refactor: move LetMatch to Desugar

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/6421

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala [hunk 1]
```diff
@@ -133,6 +133,8 @@ object WeededAst {
 
     case class ForEachYield(frags: List[ForFragment], exp: Expr, loc: SourceLocation) extends Expr
 
+    case class LetMatch(pat: Pattern, mod: Ast.Modifiers, tpe: Option[Type], exp1: Expr, exp2: Expr, loc: SourceLocation) extends Expr
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala:136]: What kind of ordering are we using here?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala:136]: Can we try to keep everything sorted in the same way, following WeededAst? :)

**Comment by magnus-madsen** [comment]
Ill merge as-is and then we can have a reordering PR :)

**Review by magnus-madsen — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala (hunk 1 above)
Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala:136]: It already does, right?
I think Weeder was wrong before I began this (LetMatch comes after ForEachYield in Weeder and Ast, but Tuple is next in Ast whereas LetRecDef is next in Weeder)

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala:136]: Anyway, we can change this after :)

**Pull request merged**